environment-api/src/test/java/com/sequenceiq/environment/api/v1/environment/model/request/aws/AwsDiskEncryptionParametersTest.java

```diff
@@ -97,19 +81,12 @@ void encryptionKeyArnValidationTest(String testCaseName, String encryptionKeyArn
                 .withEncryptionKeyArn(encryptionKeyArn)
                 .build();
 
-        Set<ConstraintViolation<AwsDiskEncryptionParameters>> constraintViolations = localValidatorFactory.validate(underTest);
+        String regex = "^arn:(aws|aws-cn|aws-us-gov):kms:[a-zA-Z0-9-]+:[0-9]+:key/[a-zA-Z0-9-]+$";
```

Comment:
the regex should come from the tested class!

Reply:
@bergerdenes I have explored from my end and It seems I could not extract just the regex from pattern annotator, Instead I should be using ValidatorFactory to validate this. Could you please help me If there is any other way that I am missing here to just extract the regex( I could not re-define a local variable of type pattern in the source file as I am already using the pattern annotator for validation  there)

Reply:
As mentioned in my other comment, I do not feel it is right to rely on regex matching in the UT, since the Hibernate validator already takes care of that for us (if used, of course).

Reply:
I think by testing the validator, we are not testing purely the pattern, but the validator as well (which is totally unnecessary as there is no new validator implemented here)
You can extract the regex by adding a static final String to your class (a package-private one) and you can use that from your unit test.

Reply:
Changed the test case to regex check as per the review comments.